lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -0,0 +1,22 @@
+"""
+This file contains implementation override of SearchFilterGenerator which will allow
+    * Filter by all courses in which the user is enrolled in
+"""
+from student.models import CourseEnrollment
+
+from search.filter_generator import SearchFilterGenerator
+
+
+class LmsSearchFilterGenerator(SearchFilterGenerator):
+    """ SearchFilterGenerator for LMS Search """
+
+    def field_dictionary(self, **kwargs):
+        """ add course if provided otherwise add courses in which the user is enrolled in """
+        field_dictionary = super(LmsSearchFilterGenerator, self).field_dictionary(**kwargs)
+        if 'user' not in kwargs or not kwargs['user']:
```

Comment:
This would be more readable written as `if not kwargs.get('user')`. Same is true of the below; it'd be more readable written as `if not kwargs.get('course_id')`.